coordinates.cpp

```diff
@@ -1 +1,58 @@
+#include "coordinates.h"
+#include <cmath> 
+#include <cstdlib>
+#include <iostream>
+#include <math.h>
+#include <utility> 
 
+using namespace std;
+
+// earth radius 
+double rad = 6371.0; 
+
+// convert decimal degrees to rads
+double degreeToRad(double deg) {
+  return (deg * M_PI / 180);
+}
+
+// convert rads to decimal degrees
+double radToDegree(double rad) {
+  return (rad * 180 / M_PI); }
+
+double Haversine(Coordinate a, Coordinate b) {
+        double lat1r, lon1r, lat2r, lon2r, latFinal, lonFinal;
+        lat1r = degreeToRad(a.first);
+        lon1r = degreeToRad(a.second);
+        lat2r = degreeToRad(b.first);
+        lon2r = degreeToRad(b.second);
+        // find the distance between lat and long 
+        latFinal = sin((lat2r - lat1r)/2);
+        lonFinal = sin((lon2r - lon1r)/2);
+        double distance = 2.0 * rad * asin(sqrt(latFinal * latFinal + cos(lat1r) * cos(lat2r) * lonFinal * lonFinal));
+        // to mi 
+        distance = distance / 1.609344;
+
+        return distance; 
+    }
+
+Coordinate SortByDistanceTo(CoordVec v1) { 
```

Comment:
Lab instructions asked for this to be sorted in-place. That would require a reference parameter.